docs/conceptual/core/codec.md

```diff
@@ -1,13 +1,181 @@
+title: SCALE Codec
 ---
 
-* What is parity scale codec?
+The SCALE (Simple Concatenated Aggregate Little-Endian) Codec is a lightweight,
+efficient, binary serialization and deserialization codec.
 
-* How does it work?
+It is designed for high-performance, copy-free encoding and decoding of data in
+resource-constrained execution contexts, like the [Substrate
+runtime](conceptual/runtime/index.md). It is not self-describing in any way and
+assumes the decoding context has all type knowledge about the encoded data.
 
-* Examples
+## SCALE for Substrate
 
-* Why did we invent our own codec?
+Substrate uses the
+[`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec), a Rust
+implementation of the SCALE Codec. This library and the SCALE codec are
+advantageous for Substrate and blockchain systems because:
 
-* What implementations exist?
\ No newline at end of file
+* It does not use Rust STD, and thus can compile to Wasm.
+* It is built to have great support in Rust for deriving codec logic for new
+  types:
+  ```
+  #[derive(Encode, Decode)]
+  ```
+
+It is important to define the encoding scheme used on Substrate rather than
+reuse an existing Rust codec library because this codec needs to be
+re-implemented on other platforms and languages that want to support
+interoperability.
+
+## Codec Definition
+
+Here you will find how the SCALE codec encodes different types.
+
+### Fixed-width Integers
+
+Basic integers are encoded using a fixed-width little-endian (LE) format.
+
+#### Example
+
+- `signed 8-bit integer 69`: `0x45`
+- `unsigned 16-bit integer 42`: `0x2a00`
+- `unsigned 32-bit integer 16777215`: `0xffffff00`
+
+### Compact/General Integers
+
+A "compact" or general integer encoding is sufficient for encoding large
+integers (up to 2**536) and is more efficient at encoding most values than the
+fixed-width version. (Though for single-byte values, the fixed-width integer is
+never worse.)
+
+It is encoded with the two least significant bits denoting the mode:
+
+- `0b00`: single-byte mode; upper six bits are the LE encoding of the value
+  (valid only for values of 0-63).
+- `0b01`: two-byte mode: upper six bits and the following byte is the LE
+  encoding of the value (valid only for values `64-(2**14-1)`).
+- `0b10`: four-byte mode: upper six bits and the following three bytes are the
+  LE encoding of the value (valid only for values `(2**14-1)-(2**30-1)`).
+- `0b11`: Big-integer mode: The upper six bits are the number of bytes
+  following, less four. The value is contained, LE encoded, in the bytes
+  following. The final (most significant) byte must be non-zero. Valid only for
+  values `(2**30-1)-(2**536-1)`.
+
+#### Example
+
+- `unsigned integer 0`: `0x00`
+- `unsigned integer 1`: `0x04`
+- `unsigned integer 42`: `0xa8`
+- `unsigned integer 69`: `0x1501`
+
+Error:
+
+- ~~`0x0100`: Zero encoded in mode 1~~
+
+### Boolean
+
+Boolean values are encoded using the least significant bit (LSB) of a single
```

Comment:
No need to introduce acronyms unless we refer to the concept again with the acronym. If I'm missing some use of LSB elsewhere in this doc, ignore this...
```suggestion
Boolean values are encoded using the least significant bit of a single
```